kafka-ui-api/src/main/java/com/provectus/kafka/ui/cluster/service/KafkaConnectService.java

```diff
@@ -0,0 +1,227 @@
+package com.provectus.kafka.ui.cluster.service;
+
+import com.provectus.kafka.ui.cluster.client.KafkaConnectClients;
+import com.provectus.kafka.ui.cluster.exception.NotFoundException;
+import com.provectus.kafka.ui.cluster.exception.RebalanceInProgressException;
+import com.provectus.kafka.ui.cluster.exception.ValidationException;
+import com.provectus.kafka.ui.cluster.mapper.ClusterMapper;
+import com.provectus.kafka.ui.cluster.mapper.KafkaConnectMapper;
+import com.provectus.kafka.ui.cluster.model.ClustersStorage;
+import com.provectus.kafka.ui.cluster.model.KafkaCluster;
+import com.provectus.kafka.ui.cluster.model.KafkaConnectCluster;
+import com.provectus.kafka.ui.model.*;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+import org.springframework.stereotype.Service;
+import org.springframework.web.reactive.function.client.WebClientResponseException;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import reactor.retry.Retry;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+@Service
+@Log4j2
+@RequiredArgsConstructor
+public class KafkaConnectService {
+    private static final int MAX_RETRIES = 5;
+
+    private final ClustersStorage clustersStorage;
+    private final ClusterMapper clusterMapper;
+    private final KafkaConnectMapper kafkaConnectMapper;
+
+    public Mono<Flux<Connect>> getConnects(String clusterName) {
+        return Mono.just(
+                Flux.fromIterable(clustersStorage.getClusterByName(clusterName)
+                        .map(KafkaCluster::getKafkaConnect).stream()
+                        .flatMap(Collection::stream)
+                        .map(clusterMapper::toKafkaConnect)
+                        .collect(Collectors.toList())
+                )
+        );
+    }
+
+
+    public Flux<String> getConnectors(String clusterName, String connectName) {
+        return getConnectAddress(clusterName, connectName)
+                .flatMapMany(connect -> withRetryOnConflict(
+                        KafkaConnectClients.withBaseUrl(connect).getConnectors()
+                        ).doOnError(log::error)
+                );
+    }
+
+    public Mono<Connector> createConnector(String clusterName, String connectName, Mono<NewConnector> connector) {
+        return getConnectAddress(clusterName, connectName)
+                .flatMap(connect ->
+                        connector.cache()
```

Comment:
Why do you cache request? may be it's better to flatmap it?

Reply:
It was needed in earlier version of this service, removed